packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -597,19 +595,21 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     if (this.props.disabled && (ev.which === KeyCodes.enter || ev.which === KeyCodes.space)) {
       ev.preventDefault();
       ev.stopPropagation();
+    } else if (!this.props.disabled && this.props.menuProps) {
+      this._onMenuKeyDown(ev);
     } else if (!this.props.disabled && this.props.onKeyDown !== undefined) {
       this.props.onKeyDown(ev); // not cancelling event because it's not disabled
     }
   };
 
   private _onKeyUp = (ev: React.KeyboardEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
-    if (!this.props.disabled && this.props.onKeyUp !== undefined) {
+    if (!this.props.disabled && this.props.onKeyUp !== undefined && this.props.menuProps) {
```

Comment:
why does the existence of menuProps determine if the keyUp callback should get called? Won't this always be false for non-ContextualMenuButtons?

Reply:
yes actually just like for the mouse up/mouse down methods, there's no need for menuProps check here or in keydown/keypress